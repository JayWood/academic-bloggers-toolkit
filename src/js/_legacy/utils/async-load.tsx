// TODO: deprecate this file
import React from 'react';

interface Props {
    component: Promise<React.ComponentClass<any>>;
    [k: string]: any;
}

interface State {
    component: React.ComponentClass<any> | null;
}

export default class AsyncLoad extends React.PureComponent<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            component: null,
        };
        this.onResolve();
    }

    render(): JSX.Element | null {
        const Component = this.state.component;
        const { component, ...props } = this.props;
        return Component === null ? null : <Component {...props} />;
    }

    private async onResolve(): Promise<void> {
        this.props.component.then(component => {
            this.setState(prevState => ({ ...prevState, component }));
        });
    }
}
