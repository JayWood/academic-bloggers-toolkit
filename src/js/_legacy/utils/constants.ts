export enum MenuActionType {
    CHANGE_STYLE = 'CHANGE_STYLE',
    DESTROY_PROCESSOR = 'DESTROY_PROCESSOR',
    INSERT_STATIC_BIBLIOGRAPHY = 'INSERT_STATIC_BIBLIOGRAPHY',
    OPEN_IMPORT_DIALOG = 'OPEN_IMPORT_DIALOG',
    REFRESH_PROCESSOR = 'REFRESH_PROCESSOR',
}

interface LocaleMapper {
    [k: string]: string | undefined;
}

/**
 * This object converts the locale names in wordpress or PubMed (keys) to the locales
 * in CSL (values). If CSL doesn't have a locale for a given WordPress locale,
 * then undefined is used (which will default to en-US)
 */
export const localeMapper: LocaleMapper = {
    af: 'af-ZA',
    'af-ZA': 'af-ZA',
    afr: 'af-ZA',
    ar: 'ar',
    'ar-AR': 'ar',
    ara: 'ar',
    arq: 'ar',
    ary: 'ar',
    az: 'tr-TR',
    az_TR: 'tr-TR',
    'bg-BG': 'bg-BG',
    bg_BG: 'bg-BG',
    bul: 'bg-BG',
    ca: 'ca-AD',
    'ca-AD': 'ca-AD',
    cat: 'ca-AD',
    chi: 'zh-CN',
    'cs-CZ': 'cs-CZ',
    cs_CZ: 'cs-CZ',
    cy: 'cy-GB',
    cze: 'cs-CZ',
    'da-DK': 'da-DK',
    da_DK: 'da-DK',
    dan: 'da-DK',
    'de-AT': 'de-AT',
    'de-CH': 'de-CH',
    'de-DE': 'de-DE',
    de_CH: 'de-CH',
    de_DE: 'de-DE',
    dut: 'nl-NL',
    el: 'el-GR',
    'el-GR': 'el-GR',
    'en-GB': 'en-GB',
    'en-US': 'en-US',
    en_GB: 'en-GB',
    'es-AR': 'es-ES',
    'es-CL': 'es-CL',
    'es-ES': 'es-ES',
    'es-MX': 'es-MX',
    es_AR: 'es-ES',
    es_CL: 'es-CL',
    es_CO: 'es-CL',
    es_ES: 'es-ES',
    es_GT: 'es-ES',
    es_MX: 'es-MX',
    es_PE: 'es-CL',
    es_PR: 'es-CL',
    es_VE: 'es-CL',
    est: 'et-ET',
    et: 'et-ET',
    'et-ET': 'et-ET',
    eu: 'eu',
    'fa-IR': 'fa-IR',
    fa_AF: 'fa-IR',
    fa_IR: 'fa-IR',
    fi: 'fi-FI',
    'fi-FI': 'fi-FI',
    fin: 'fi-FI',
    'fr-CA': 'fr-CA',
    'fr-FR': 'fr-FR',
    fr_BE: 'fr-FR',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    fre: 'fr-FR',
    ger: 'de-DE',
    gre: 'el-GR',
    gsw: 'de-CH',
    'he-IL': 'he-IL',
    he_IL: 'he-IL',
    heb: 'he-IL',
    hr: 'hr-HR',
    'hr-HR': 'hr-HR',
    hrv: 'hr-HR',
    'hu-HU': 'hu-HU',
    hu_HU: 'hu-HU',
    hun: 'hu-HU',
    ice: 'is-IS',
    'id-ID': 'id-ID',
    id_ID: 'id-ID',
    ind: 'id-ID',
    'is-IS': 'is-IS',
    is_IS: 'is-IS',
    'it-IT': 'it-IT',
    it_IT: 'it-IT',
    ita: 'it-IT',
    ja: 'ja-JP',
    'ja-JP': 'ja-JP',
    jpn: 'ja-JP',
    km: 'km-KH',
    'km-KH': 'km-KH',
    'ko-KR': 'ko-KR',
    ko_KR: 'ko-KR',
    kor: 'ko-KR',
    lav: 'lv-LV',
    lb_LU: 'lt-LT',
    lit: 'lt-LT',
    'lt-LT': 'lt-LT',
    lt_LT: 'lt-LT',
    lv: 'lv-LV',
    'lv-LV': 'lv-LV',
    mn: 'mn-MN',
    'mn-MN': 'mn-MN',
    'nb-NO': 'nb-NO',
    nb_NO: 'nb-NO',
    nl: 'nl-NL',
    'nl-NL': 'nl-NL',
    nl_BE: 'nl-NL',
    nl_NL: 'nl-NL',
    'nn-NO': 'nn-NO',
    nn_NO: 'nn-NO',
    nor: 'nn-NO',
    per: 'fa-IR',
    'pl-PL': 'pl-PL',
    pl_PL: 'pl-PL',
    pol: 'pl-PL',
    por: 'pt-PT',
    'pt-BR': 'pt-BR',
    'pt-PT': 'pt-PT',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    'ro-RO': 'ro-RO',
    ro_RO: 'ro-RO',
    'ru-RU': 'ru-RU',
    ru_RU: 'ru-RU',
    rum: 'ro-RO',
    rus: 'ru-RU',
    'sk-SK': 'sk-SK',
    sk_SK: 'sk-SK',
    'sl-SI': 'sl-SI',
    sl_SI: 'sl-SI',
    slo: 'sk-SK',
    slv: 'sl-SI',
    spa: 'es-ES',
    'sr-RS': 'sr-RS',
    sr_RS: 'sr-RS',
    srp: 'sr-RS',
    'sv-SE': 'sv-SE',
    sv_SE: 'sv-SE',
    swe: 'sv-SE',
    th: 'th-TH',
    'th-TH': 'th-TH',
    tha: 'th-TH',
    'tr-TR': 'tr-TR',
    tr_TR: 'tr-TR',
    tur: 'tr-TR',
    uk: 'uk-UA',
    'uk-UA': 'uk-UA',
    ukr: 'uk-UA',
    vi: 'vi-VN',
    'vi-VN': 'vi-VN',
    vie: 'vi-VN',
    'zh-CN': 'zh-CN',
    'zh-TW': 'zh-TW',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-CN',
    zh_TW: 'zh-TW',
};
