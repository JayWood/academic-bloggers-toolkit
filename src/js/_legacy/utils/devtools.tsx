import AsyncLoad from '_legacy/utils/async-load';
import React from 'react';

export default function devtool(props: any): any {
    if (process.env.NODE_ENV !== 'production') {
        return (
            <AsyncLoad
                component={import('mobx-react-devtools').then(
                    mod => mod.default,
                )}
                {...props}
            />
        );
    }
    return (): null => null;
}

export function configureDevtool(options: {
    graphEnabled?: boolean;
    logEnabled?: boolean;
    updatesEnabled?: boolean;
    logFilter?(p: any): boolean;
}): void {
    if (process.env.NODE_ENV !== 'production') {
        const cdt = require('mobx-react-devtools').configureDevtool;
        cdt(options);
    }
}
