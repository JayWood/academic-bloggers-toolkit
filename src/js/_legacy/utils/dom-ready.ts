/**
 * A better jQuery `$(document).ready()`
 */
export default async function domReady(): Promise<{}> {
    return new Promise<{}>(
        (resolve): void => {
            if (document.readyState !== 'loading') {
                return resolve();
            } else {
                document.addEventListener('DOMContentLoaded', resolve);
            }
        },
    );
}
