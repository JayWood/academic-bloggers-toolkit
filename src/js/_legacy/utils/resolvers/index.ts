export { getFromDOI } from './doi';
export { getFromPubmed, pubmedQuery } from './pubmed';
export { getFromISBN } from './isbn';
export { getFromURL } from './url';

type StringFields = { [k in CSL.StandardFieldKey | CSL.DateFieldKey]?: string };
type PersonFields = ABT.Contributor[];
export interface AutociteResponse {
    fields: StringFields;
    people: PersonFields;
}
