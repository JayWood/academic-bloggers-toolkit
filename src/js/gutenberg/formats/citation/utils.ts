interface CCHProps {
    id: string;
    items: string[];
}

export function createCitationHtml({ id, items }: CCHProps): string {
    const TERMINATOR = '&#65279;';
    const citation = document.createElement('span');
    citation.className = 'abt-citation';
    citation.dataset.id = id;
    citation.dataset.items = JSON.stringify(items);
    citation.contentEditable = 'false';
    return `${citation.outerHTML}${TERMINATOR}`;
}
