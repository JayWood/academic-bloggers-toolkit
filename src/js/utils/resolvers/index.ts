import * as doi from './doi';
import * as isbn from './isbn';
import * as pubmed from './pubmed';
import * as url from './url';

export { doi, isbn, pubmed, url };
