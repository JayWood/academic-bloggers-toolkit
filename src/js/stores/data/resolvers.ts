import { Actions } from './constants';
import { fetchCitationStyles } from './controls';

export function* getCitationStyles() {
    const styles = yield fetchCitationStyles();
    return {
        type: Actions.SET_CITATION_STYLES,
        styles,
    };
}
