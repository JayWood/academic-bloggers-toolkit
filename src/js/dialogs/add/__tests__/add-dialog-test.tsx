jest.mock('utils/resolvers/');
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as React from 'react';

import { BookMeta, getFromISBN, getFromURL } from 'utils/resolvers/';
import AddDialog from '../';

const mocks = {
    getFromURL: getFromURL as jest.Mock<{}>,
    getFromISBN: getFromISBN as jest.Mock<{}>,
    onSubmit: jest.fn(),
};

const setup = () => {
    const component = shallow(<AddDialog onSubmit={mocks.onSubmit} />);
    const instance = component.instance() as AddDialog;
    return {
        component,
        instance,
    };
};

describe('<AddDialog />', () => {
    beforeEach(() => jest.resetAllMocks());
    it('should match snapshot', () => {
        const { component, instance } = setup();

        // Add with identifiers
        expect(toJSON(component)).toMatchSnapshot();

        // Add manually
        instance.addManually.set(true);
        component.update().render();
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should handle pubmed dialog submit', () => {
        const { component, instance } = setup();
        expect(instance.identifierList.get()).toBe('');
        const ButtonRow = component.find('ButtonRow');

        ButtonRow.simulate('pubmedDialogSubmit', '12345');
        expect(instance.identifierList.get()).toBe('12345');

        ButtonRow.simulate('pubmedDialogSubmit', '54321');
        expect(instance.identifierList.get()).toBe('12345,54321');

        // TODO: Handle duplicates
    });
    it('should handle identifier change', () => {
        const { component, instance } = setup();
        expect(instance.identifierList.get()).toBe('');
        const input = component.find('IdentifierInput');
        input.simulate('change', { currentTarget: { value: 'foo' } });
        expect(instance.identifierList.get()).toBe('foo');
    });
    it('should handle type change', () => {
        const { component, instance } = setup();
        instance.addManually.set(true);
        instance.manualData.set('randomThing', 'random value');
        instance.people.push({ family: 'Smith', given: 'Bob', type: 'editor' });
        expect([...instance.manualData.entries()]).toEqual([
            ['type', 'webpage'],
            ['randomThing', 'random value'],
        ]);
        expect(instance.people.length).toBe(2);
        const manualEntryContainer = component.find('ManualEntryContainer');
        manualEntryContainer.simulate('typeChange', 'book');
        expect([...instance.manualData.entries()]).toEqual([['type', 'book']]);
        expect(instance.people.length).toBe(1);
    });
    it('should toggle attach inline', () => {
        const { component, instance } = setup();
        expect(instance.attachInline.get()).toBe(true);

        const buttonRow = component.find('ButtonRow');
        buttonRow.simulate('attachInlineToggle');
        expect(instance.attachInline.get()).toBe(false);
    });
    it('should toggle loading state', () => {
        const { instance } = setup();
        expect(instance.isLoading.get()).toBe(false);

        instance.toggleLoadingState();
        expect(instance.isLoading.get()).toBe(true);

        instance.toggleLoadingState(true);
        expect(instance.isLoading.get()).toBe(true);

        instance.toggleLoadingState();
        expect(instance.isLoading.get()).toBe(false);
    });
    it('should toggle add manually state', () => {
        const { component, instance } = setup();
        expect(instance.addManually.get()).toBe(false);

        const buttonRow = component.find('ButtonRow');
        buttonRow.simulate('toggleManual');

        expect(instance.addManually.get()).toBe(true);
    });
    it('should handle submit', () => {
        const { component } = setup();
        const form = component.find('form');
        const preventDefault = jest.fn();
        form.simulate('submit', { preventDefault });
        expect(preventDefault).toHaveBeenCalledTimes(1);
        expect(mocks.onSubmit).toHaveBeenCalledTimes(1);
    });
    it('should capture input field', () => {
        const { instance } = setup();
        const input = document.createElement('input');
        const focusMock = jest.fn();
        input.focus = focusMock;
        instance.captureInputField(input);
        instance.appendPMID('12345');
        expect(focusMock).toHaveBeenCalled();
    });
    it('should skip capture of input field if null', () => {
        const { instance } = setup();
        instance.captureInputField(null);
        expect(instance.appendPMID).not.toThrow();
    });
    describe('autocite handler tests', () => {
        let component: any;
        let instance: any;
        let manualEntryContainer: any;
        const today = new Date().toISOString().substr(0, 10).replace(/-/g, '/');
        beforeEach(() => {
            jest.resetAllMocks();

            ({ component, instance } = setup());
            expect([...instance.manualData.entries()]).toEqual([['type', 'webpage']]);

            instance.addManually.set(true);
            component.update();

            manualEntryContainer = component.find('ManualEntryContainer');
        });
        test('webpage', async () => {
            const data: any = {
                accessed: '2003-01-02T05:00:00.000Z​​​​​',
                authors: [
                    {
                        firstname: 'John',
                        lastname: 'Doe',
                    },
                    {
                        firstname: 'Jane',
                        lastname: 'Smith',
                    },
                    {
                        firstname: undefined,
                        lastname: undefined,
                    },
                ],
                content_title: 'Test Title',
                issued: '2003-01-02T05:00:00.000Z​​​​​',
                site_title: 'Google',
                url: 'www.google.com',
            };
            const expected = [
                [
                    ['type', 'webpage'],
                    ['URL', 'www.google.com'],
                    ['accessed', '2003/01/02'],
                    ['container-title', 'Google'],
                    ['issued', '2003/01/02'],
                    ['title', 'Test Title'],
                ],
            ];
            mocks.getFromURL.mockReturnValueOnce(Promise.resolve(data));
            await manualEntryContainer.simulate('autoCite', 'webpage', 'www.google.com');
            expect([instance.manualData.entries()]).toEqual(expected);
        });
        test('book', async () => {
            instance.manualData.set('type', 'book');
            const data: BookMeta = {
                title: 'Test Title',
                'number-of-pages': '100',
                publisher: 'Test Publisher',
                issued: '2003/01/02',
                authors: [
                    {
                        family: 'Smith',
                        given: 'John',
                        type: 'author',
                    },
                    {
                        family: 'Doe',
                        given: 'Jane',
                        type: 'author',
                    },
                ],
            };
            const expected = [
                [
                    ['type', 'book'],
                    ['accessed', today],
                    ['issued', '2003/01/02'],
                    ['number-of-pages', '100'],
                    ['publisher', 'Test Publisher'],
                    ['title', 'Test Title'],
                ],
            ];
            mocks.getFromISBN.mockReturnValueOnce(Promise.resolve(data));
            await manualEntryContainer.simulate('autoCite', 'book', '1234567890');
            expect([instance.manualData.entries()]).toEqual(expected);
        });
        test('chapter', async () => {
            instance.manualData.set('type', 'chapter');
            const data: BookMeta = {
                title: 'Test Title',
                'number-of-pages': '100',
                publisher: 'Test Publisher',
                issued: '2003/01/02',
                authors: [
                    {
                        family: 'Smith',
                        given: 'John',
                        type: 'author',
                    },
                    {
                        family: 'Doe',
                        given: 'Jane',
                        type: 'author',
                    },
                ],
            };
            const expected = [
                [
                    ['type', 'chapter'],
                    ['accessed', today],
                    ['issued', '2003/01/02'],
                    ['number-of-pages', '100'],
                    ['publisher', 'Test Publisher'],
                    ['container-title', 'Test Title'],
                ],
            ];
            mocks.getFromISBN.mockReturnValueOnce(Promise.resolve(data));
            await manualEntryContainer.simulate('autoCite', 'chapter', '1234567890');
            expect([instance.manualData.entries()]).toEqual(expected);
        });
        it('should handle errors', async () => {
            mocks.getFromURL.mockReturnValueOnce(Promise.reject(new Error('Test error handling')));
            await manualEntryContainer.simulate('autoCite', 'webpage', 'www.google.com');
            mocks.getFromURL.mockReturnValueOnce(Promise.reject({}));
            await manualEntryContainer.simulate('autoCite', 'webpage', 'www.google.com');
            expect(true).toBe(true);
        });
    });
});
