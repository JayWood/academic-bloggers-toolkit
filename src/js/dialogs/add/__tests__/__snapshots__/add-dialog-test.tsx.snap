// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddDialog /> should match snapshot 1`] = `
<div>
  <form
    id="add-reference"
    onSubmit={[Function]}
  >
    <IdentifierInput
      fieldRef={[Function]}
      identifierList={""}
      onChange={[Function]}
    />
  </form>
  <ButtonRow
    addManually={false}
    attachInline={true}
    isLoading={false}
    onAttachInlineToggle={[Function]}
    onSearchPubmedClick={[Function]}
    onToggleManual={[Function]}
  />
  <style
    jsx={true}
  >
    
                    div {
                        position: relative;
                    }
                
  </style>
</div>
`;

exports[`<AddDialog /> should match snapshot 2`] = `
<div>
  <form
    id="add-reference"
    onSubmit={[Function]}
  >
    <ManualEntryContainer
      errorMessage={""}
      manualData={
        Object {
          "type": "webpage",
        }
      }
      onAutoCite={[Function]}
      onTypeChange={[Function]}
      people={
        Array [
          Object {
            "family": "",
            "given": "",
            "type": "author",
          },
        ]
      }
    />
  </form>
  <ButtonRow
    addManually={true}
    attachInline={true}
    isLoading={false}
    onAttachInlineToggle={[Function]}
    onSearchPubmedClick={[Function]}
    onToggleManual={[Function]}
  />
  <style
    jsx={true}
  >
    
                    div {
                        position: relative;
                    }
                
  </style>
</div>
`;

exports[`<AddDialog /> should open pubmed dialog 1`] = `
<div>
  <form
    id="add-reference"
    onSubmit={[Function]}
  >
    <IdentifierInput
      fieldRef={[Function]}
      identifierList={""}
      onChange={[Function]}
    />
  </form>
  <ButtonRow
    addManually={false}
    attachInline={true}
    isLoading={false}
    onAttachInlineToggle={[Function]}
    onSearchPubmedClick={[Function]}
    onToggleManual={[Function]}
  />
  <style
    jsx={true}
  >
    
                    div {
                        position: relative;
                    }
                
  </style>
</div>
`;

exports[`<AddDialog /> should open pubmed dialog 2`] = `
<div>
  <Container
    currentDialog={"PUBMED"}
    onClose={[Function]}
    overlayOpacity={0.2}
    title="Search PubMed"
    width={600}
  >
    <PubmedDialog
      onSubmit={[Function]}
    />
  </Container>
  <form
    id="add-reference"
    onSubmit={[Function]}
  >
    <IdentifierInput
      fieldRef={[Function]}
      identifierList={""}
      onChange={[Function]}
    />
  </form>
  <ButtonRow
    addManually={false}
    attachInline={true}
    isLoading={false}
    onAttachInlineToggle={[Function]}
    onSearchPubmedClick={[Function]}
    onToggleManual={[Function]}
  />
  <style
    jsx={true}
  >
    
                    div {
                        position: relative;
                    }
                
  </style>
</div>
`;

exports[`<AddDialog /> should open pubmed dialog 3`] = `
<div>
  <Container
    currentDialog={"PUBMED"}
    onClose={[Function]}
    overlayOpacity={0.2}
    title="Search PubMed"
    width={600}
  >
    <PubmedDialog
      onSubmit={[Function]}
    />
  </Container>
  <form
    id="add-reference"
    onSubmit={[Function]}
  >
    <IdentifierInput
      fieldRef={[Function]}
      identifierList={""}
      onChange={[Function]}
    />
  </form>
  <ButtonRow
    addManually={false}
    attachInline={true}
    isLoading={false}
    onAttachInlineToggle={[Function]}
    onSearchPubmedClick={[Function]}
    onToggleManual={[Function]}
  />
  <style
    jsx={true}
  >
    
                    div {
                        position: relative;
                    }
                
  </style>
</div>
`;
